windows/devices/uwb/UwbCxAdapter.cxx

```diff
@@ -0,0 +1,239 @@
+ * Taken from the FiRa UCI spec Table 29:APP Configuration IDs
+ *
+ */
+const std::unordered_map<UWB_APP_CONFIG_PARAM_TYPE, unsigned int> appConfigParamLengths{
+    { UWB_APP_CONFIG_PARAM_TYPE_DEVICE_TYPE, 1 },
+    { UWB_APP_CONFIG_PARAM_TYPE_RANGING_ROUND_USAGE, 1 },
+    { UWB_APP_CONFIG_PARAM_TYPE_STS_CONFIG, 1 },
+    { UWB_APP_CONFIG_PARAM_TYPE_MULTI_NODE_MODE, 1 },
+    { UWB_APP_CONFIG_PARAM_TYPE_CHANNEL_NUMBER, 1 },
+    { UWB_APP_CONFIG_PARAM_TYPE_NUMBER_OF_CONTROLEES, 1 },
+    { UWB_APP_CONFIG_PARAM_TYPE_DEVICE_MAC_ADDRESS, 0 },
+    { UWB_APP_CONFIG_PARAM_TYPE_DST_MAC_ADDRESS, 0 },
+    { UWB_APP_CONFIG_PARAM_TYPE_SLOT_DURATION, 2 },
+    { UWB_APP_CONFIG_PARAM_TYPE_RANGING_INTERVAL, 4 },
+    { UWB_APP_CONFIG_PARAM_TYPE_STS_INDEX, 4 },
+    { UWB_APP_CONFIG_PARAM_TYPE_MAC_FCS_TYPE, 1 },
+    { UWB_APP_CONFIG_PARAM_TYPE_RANGING_ROUND_CONTROL, 1 },
+    { UWB_APP_CONFIG_PARAM_TYPE_AOA_RESULT_REQ, 1 },
+    { UWB_APP_CONFIG_PARAM_TYPE_RANGE_DATA_NTF_CONFIG, 1 },
+    { UWB_APP_CONFIG_PARAM_TYPE_RANGE_DATA_NTF_PROXIMITY_NEAR, 2 },
+    { UWB_APP_CONFIG_PARAM_TYPE_RANGE_DATA_NTF_PROXIMITY_FAR, 2 },
+    { UWB_APP_CONFIG_PARAM_TYPE_DEVICE_ROLE, 1 },
+    { UWB_APP_CONFIG_PARAM_TYPE_RFRAME_CONFIG, 1 },
+    { UWB_APP_CONFIG_PARAM_TYPE_PREAMBLE_CODE_INDEX, 1 },
+    { UWB_APP_CONFIG_PARAM_TYPE_SFD_ID, 1 },
+    { UWB_APP_CONFIG_PARAM_TYPE_PSDU_DATA_RATE, 1 },
+    { UWB_APP_CONFIG_PARAM_TYPE_PREAMBLE_DURATION, 1 },
+    { UWB_APP_CONFIG_PARAM_TYPE_RANGING_TIME_STRUCT, 1 },
+    { UWB_APP_CONFIG_PARAM_TYPE_SLOTS_PER_RR, 1 },
+    { UWB_APP_CONFIG_PARAM_TYPE_TX_ADAPTIVE_PAYLOAD_POWER, 1 },
+    { UWB_APP_CONFIG_PARAM_TYPE_RESPONDER_SLOT_INDEX, 1 },
+    { UWB_APP_CONFIG_PARAM_TYPE_PRF_MODE, 1 },
+    { UWB_APP_CONFIG_PARAM_TYPE_SCHEDULED_MODE, 1 },
+    { UWB_APP_CONFIG_PARAM_TYPE_KEY_ROTATION, 1 },
+    { UWB_APP_CONFIG_PARAM_TYPE_KEY_ROTATION_RATE, 1 },
+    { UWB_APP_CONFIG_PARAM_TYPE_SESSION_PRIORITY, 1 },
+    { UWB_APP_CONFIG_PARAM_TYPE_MAC_ADDRESS_MODE, 1 },
+    { UWB_APP_CONFIG_PARAM_TYPE_VENDOR_ID, 2 },
+    { UWB_APP_CONFIG_PARAM_TYPE_STATIC_STS_IV, 6 },
+    { UWB_APP_CONFIG_PARAM_TYPE_NUMBER_OF_STS_SEGMENTS, 1 },
+    { UWB_APP_CONFIG_PARAM_TYPE_MAX_RR_RETRY, 2 },
+    { UWB_APP_CONFIG_PARAM_TYPE_UWB_INITIATION_TIME, 4 },
+    { UWB_APP_CONFIG_PARAM_TYPE_HOPPING_MODE, 1 },
+    { UWB_APP_CONFIG_PARAM_TYPE_BLOCK_STRIDE_LENGTH, 1 },
+    { UWB_APP_CONFIG_PARAM_TYPE_RESULT_REPORT_CONFIG, 1 },
+    { UWB_APP_CONFIG_PARAM_TYPE_IN_BAND_TERMINATION_ATTEMPT_COUNT, 1 },
+    { UWB_APP_CONFIG_PARAM_TYPE_SUB_SESSION_ID, 4 },
+    { UWB_APP_CONFIG_PARAM_TYPE_BPRF_PHR_DATA_RATE, 1 },
+    { UWB_APP_CONFIG_PARAM_TYPE_MAX_NUMBER_OF_MEASUREMENTS, 2 },
+    { UWB_APP_CONFIG_PARAM_TYPE_STS_LENGTH, 1 }
+};
+
+const std::unordered_map<
+    ::uwb::protocol::fira::UwbConfiguration::ParameterTag,
+    UWB_APP_CONFIG_PARAM_TYPE>
+    service_ddi_map{
```

Comment:
This map always gets used as an argument to the `UwbAppConfigurationParameter` constructor. Might it make sense to have the `UwbAppConfigurationParameter` constructor use the map directly and simplify its interface to take the parameter tag instead? Eg. Change this:

```
explicit UwbAppConfigurationParameter(const PropertyT& value, UWB_APP_CONFIG_PARAM_TYPE parameterType, size_t parameterSize = sizeof(PropertyT)) 
```

to this:
```
explicit UwbAppConfigurationParameter(const PropertyT& value, UwbConfiguration::ParameterTag parameterTag, size_t parameterSize = sizeof(PropertyT)) 
```

Then the calling code can do this:
```
builder.Push(UwbAppConfigurationParameter{ static_cast<uint8_t>(arg), ptag }));
```

The caller is then required to know absolutely nothing about the UwbCx details like `UWB_APP_CONFIG_PARAM_TYPE `, which is especially nice since this adapter is supposed to help working with the UwbCx.

Reply:
I agree with this